gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_model.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:agentapi/agentapi.dart';
+import 'package:dart_either/dart_either.dart';
+import 'package:flutter/foundation.dart';
+import 'package:p4w_ms_store/p4w_ms_store.dart';
+import 'package:url_launcher/url_launcher.dart';
+import '/core/agent_api_client.dart';
+import '/core/environment.dart';
+import '/core/pro_token.dart';
+
+class SubscribeNowModel {
+  final AgentApiClient client;
+  bool? _isPurchaseAllowed;
+  SubscribeNowModel(this.client);
+
+  Future<SubscriptionInfo> applyProToken(ProToken token) async {
+    await client.applyProToken(token.value);
+    return client.subscriptionInfo();
+  }
+
+  void launchProWebPage() {
+    launchUrl(Uri.parse('https://ubuntu.com/pro'));
```

Comment:
A small enhancement since we're passing around.

```suggestion
    launchUrl(Uri.https('ubuntu.com/pro'));
```